feat: add database H2

MikhailViktorov opened the pull request.

MikhailViktorov committed:
feat: add database H2

MikhailViktorov committed:
feat: add database tests

MikhailViktorov committed:
fix: encoding

andrew-larkin commented on src/main/java/ru/yandex/practicum/filmorate/controller/GenreController.java:
```diff
@@ -0,0 +1,33 @@
+@RequestMapping("/genres")
+@AllArgsConstructor
+public class GenreController {
+    FilmService filmService;
```
модификатор доступа

andrew-larkin commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -1,16 +1,18 @@
+import java.util.Set;
 
 
 @Data
```
т.к. мы добавили БД, а данный класс стал моделью-сущностью фактической таблицы, нам стоит отказаться от использования @Data, которая некорректно рассчитывает хэш - ее можно заменить на несколько аннотаций @Getter, @Setter и т.д.

andrew-larkin commented on src/main/java/ru/yandex/practicum/filmorate/model/FilmMpa.java:
```diff
@@ -0,0 +1,11 @@
+@Data
+public class FilmMpa {
+    @NotNull
+    private int id;
```
т.к. тут примитив, он не может принимать null в качестве значения. поэтому тут 2 варианта: 1) отказаться от @NotNull; 2) использовать Integer

andrew-larkin reviewed (changes requested):
Привет! Хорошая реализация, хорошие запросы к БД и по архитектуре вопросов нет. Есть пара корректировок - их сделать и приложение будет полностью готово.

MikhailViktorov committed:
fix:
- access modifier filmService in GenreController.java
-delete @Data annotation in model classes
-delete @NotNull annotation in FilmMpa.java
